File: Tarea3_Recorrido_Insercion_Busqueda_01-09-25/LEM_Recorrido_Insercion_Busqueda.py
Nombres = ["Luis", "Romina", "Chema", "Carlitos"]

print("Recorrido por el array.")
for j in range(len(Nombres)):
    print(Nombres[j], end="  ")

print("\nInsertar Nombre")
insert = input("\nEscribe el nombre a insertar ") 
Nombres.insert(1, insert)
print()

for j in range(len(Nombres)):
    print(Nombres[j], end="  ")

print("Busqueda Lineal")
buscarnom = input("Escribe el nombre que quieres buscar: ") 
exist = False 

for j in Nombres:
    if j == buscarnom:
        print("Si existe el nombre")
        exist = True
        break
    
if exist == False:
    print("No existe el nombre")

File: Tarea3_Recorrido_Insercion_Busqueda_01-09-25/LEM_Recorrido_Insercion_Busqueda.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        List<string> Nombres = new List<string> { "Luis", "Romina", "Chema", "Carlitos" };

        Console.WriteLine("Recorrido por el array.");
        for (int j = 0; j < Nombres.Count; j++)
        {
            Console.Write(Nombres[j] + "  ");
        }
        Console.WriteLine();

        Console.WriteLine("\nInsertar Nombre");
        Console.Write("Escribe el nombre a insertar: ");
        string insert = Console.ReadLine();
        Nombres.Insert(1, insert);

        for (int j = 0; j < Nombres.Count; j++)
        {
            Console.Write(Nombres[j] + "  ");
        }
        Console.WriteLine();

        Console.WriteLine("Busqueda Lineal");
        Console.Write("Escribe el nombre que quieres buscar: ");
        string buscarnom = Console.ReadLine();

        bool exist = false;
        foreach (string j in Nombres)
        {
            if (j == buscarnom)
            {
                Console.WriteLine("Si existe el nombre");
                exist = true;
                break;
            }
        }

        if (!exist)
        {
            Console.WriteLine("No existe el nombre");
        }
    }
}
